libutils/json.c

```diff
@@ -340,6 +335,7 @@ static int JsonContainerCompare(
 
     if (type_a != type_b)
     {
+        Log(LOG_LEVEL_DEBUG, "JsonContainerCompare() fails, type %d not equal to type %d", type_a, type_b);
```

Comment:
You may consider logging the string representation of the type, e.g., `null`, `true`, `false`, `number`, `object`, `array`. But not strictly necessary in a debug log message in my opinion.

Reply:
This is the container case, so should only be array or object right? I can add JsonContainerTypeToString() and use that...

Reply:
got it

```
test_compare_container_type_mismatch: Starting test
   debug: JsonContainerCompare() fails, container type 'object' not equal to container type 'array'
   debug: JsonObjectCompare() fails for key 'key', children are not equal
```